src/dataflow-types/src/reconciliation/command.rs

```diff
@@ -0,0 +1,268 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+
+//! Functionality to reconcile commands between a COMPUTE controller and a COMPUTE instance.
+//!
+//! The [ComputeCommandReconcile] struct implements [GenericClient],
+//! which allow the controller to reconnect after restarts. It maintains enough state to
+//! get a newly connected instance up-to-date and matches existing installed objects with
+//! what the controller wants to provide.
+//!
+//! [ComputeCommandReconcile] is designed to live in a COMPUTE instance and liberally uses `assert` to
```

Comment:
my only review comment on this pr is that, after reading `reconciliation.md`, I found this sentence confusing, before reading the code. I think what was not clear, from just the docs, which I think should be clear is the following set of "truths":

- The `controller` is a thick client built on top of a set of `GenericClient`'s, pointing at `compute instances", hosted in the `coord` process. (As far as I can tell `controller` is not mentioned in any other docs, but is alluded to in the code).
- The `controller"` does not need reconciliation because it stores its state and is single-instance
- Each `compute instance` holds onto a `GenericClient` that receives commands from the `controller` side of that client, and reconciliation is specifically designed for each `compute` instance to handle faults

if any of these are wrong, or somehow my reading of the docs missed that they were explained, please let me know! I think its important to have it immediately clear whats going on!